test/xrt/08_gemm_extern_vec/run.lit

```diff
@@ -0,0 +1,10 @@
+// (c) Copyright 2024 Advanced Micro Devices, Inc.
+// SPDX-License-Identifier: MIT
+
+// REQUIRES: ryzen_ai, valid_xchess_license
+
+// RUN: xchesscc_wrapper aie2 -I %aietools/include -c %S/mm.cc -o mm.o
+// RUN: %python %S/aie.py
+// RUN: %python aiecc.py --no-aiesim --aie-generate-cdo --aie-generate-ipu --no-compile-host --xclbin-name=aie.xclbin --ipu-insts-name=insts.txt aie.mlir
+// RUN: g++-13 %S/test.cpp -o test.exe -std=c++23 -Wall %xrt_flags -lrt -lstdc++ -lboost_program_options -lboost_filesystem
```

Comment:
Do we want to have the version of `g++` hardcoded there?

Reply:
It's somewhat necessary at the moment to ensure c++23 for bf16

Reply:
Oh, I see. I'm so used to clang nowadays that I thought g++ 13 was an old version. Still, I don't think we're too far from g++ 14 release, and this test should pass with both. Ideally, support for c++23 should be checked, and the test enabled or disable accordingly, for good measure.

You can get a list of supported C++ standards with something like:

```sh
gcc -v --help 2> /dev/null | grep "std=" | grep Conform | cut -d"=" -f 2 | cut -d" " -f 1 | grep c++
```

Or:
```python
def list_gcc_supported_standards():
    supported = set()
    p = subprocess.run(['g++', '-v', '--help'], capture_output=True, encoding='utf8')
    for line in p.stdout.split('\n'):
        m = re.match(r'  -std=(c\+\+[\w]+)', line)
        if m:
            supported.add(m.group(1))
    return supported
```
In python.